File: fix_db.js
const { Client } = require('pg');

const client = new Client({
  connectionString: 'postgresql://postgres.gzkohtgqtpbscczxuaaj:TJHPE@B23UOM@aws-1-ap-southeast-1.pooler.supabase.com:5432/postgres?sslmode=require',
});

async function run() {
  await client.connect();
  try {
    await client.query(`ALTER TABLE packages ADD COLUMN IF NOT EXISTS agent_id BIGINT;`);
    await client.query(`ALTER TABLE packages ADD COLUMN IF NOT EXISTS start_place VARCHAR(255);`);
    await client.query(`ALTER TABLE packages ADD COLUMN IF NOT EXISTS end_place VARCHAR(255);`);
    await client.query(`ALTER TABLE packages ADD COLUMN IF NOT EXISTS price_from DOUBLE PRECISION;`);
    await client.query(`ALTER TABLE packages ADD COLUMN IF NOT EXISTS price_to DOUBLE PRECISION;`);
    await client.query(`ALTER TABLE packages ADD COLUMN IF NOT EXISTS review_count INTEGER;`);
    await client.query(`ALTER TABLE packages ADD COLUMN IF NOT EXISTS festival_details TEXT;`);
    await client.query(`ALTER TABLE packages ADD COLUMN IF NOT EXISTS trending BOOLEAN DEFAULT FALSE;`);
    await client.query(`ALTER TABLE packages ADD COLUMN IF NOT EXISTS is_active BOOLEAN DEFAULT TRUE;`);
    await client.query(`ALTER TABLE packages ADD COLUMN IF NOT EXISTS inclusions TEXT;`);
    await client.query(`ALTER TABLE packages ADD COLUMN IF NOT EXISTS application_status VARCHAR(255) DEFAULT 'Pending';`);
    
    await client.query(`ALTER TABLE hotels ADD COLUMN IF NOT EXISTS review_count INTEGER;`);
    await client.query(`ALTER TABLE hotels ADD COLUMN IF NOT EXISTS number_of_rooms INTEGER;`);
    await client.query(`ALTER TABLE hotels ADD COLUMN IF NOT EXISTS owner_name VARCHAR(255);`);
    await client.query(`ALTER TABLE hotels ADD COLUMN IF NOT EXISTS owner_email VARCHAR(255);`);
    await client.query(`ALTER TABLE hotels ADD COLUMN IF NOT EXISTS owner_nic VARCHAR(255);`);
    await client.query(`ALTER TABLE hotels ADD COLUMN IF NOT EXISTS nic_image_url VARCHAR(255);`);
    await client.query(`ALTER TABLE hotels ADD COLUMN IF NOT EXISTS phone_number VARCHAR(255);`);
    await client.query(`ALTER TABLE hotels ADD COLUMN IF NOT EXISTS hotline_number VARCHAR(255);`);
    await client.query(`ALTER TABLE hotels ADD COLUMN IF NOT EXISTS application_status VARCHAR(255) DEFAULT 'Pending';`);

    await client.query(`ALTER TABLE bookings ADD COLUMN IF NOT EXISTS hotel_id BIGINT;`);
    await client.query(`ALTER TABLE bookings ADD COLUMN IF NOT EXISTS vehicle_id BIGINT;`);
    await client.query(`ALTER TABLE bookings ADD COLUMN IF NOT EXISTS progress INTEGER DEFAULT 0;`);
    
    console.log("Successfully added all required missing columns back to Supabase");
  } catch (e) {
    console.error(e);
  }
  await client.end();
}
run().catch(console.error);

File: fix_db.py
import psycopg2

conn = psycopg2.connect(
    host="aws-1-ap-southeast-1.pooler.supabase.com",
    port=6543,
    user="postgres.gzkohtgqtpbscczxuaaj",
    password="TJHPE@B23UOM",
    dbname="postgres",
    sslmode="require"
)
cur = conn.cursor()

# Create table
cur.execute("""
CREATE TABLE IF NOT EXISTS review_images (
    id BIGSERIAL PRIMARY KEY,
    review_id BIGINT NOT NULL,
    image_url TEXT NOT NULL,
    CONSTRAINT fk_review_images_review 
        FOREIGN KEY (review_id) REFERENCES reviews(id) ON DELETE CASCADE
);
""")

# Create index
cur.execute("CREATE INDEX IF NOT EXISTS idx_review_images_review_id ON review_images(review_id);")

# Add columns
try:
    cur.execute("ALTER TABLE reviews ADD COLUMN user_name VARCHAR(255);")
except Exception as e:
    print(f"Skipped user_name: {e}")
    conn.rollback()

try:
    cur.execute("ALTER TABLE reviews ADD COLUMN title VARCHAR(255);")
except Exception as e:
    print(f"Skipped title: {e}")
    conn.rollback()

conn.commit()
cur.close()
conn.close()
print("DB Fix applied!")
